homeassistant/components/yamaha/config_flow.py

```diff
@@ -0,0 +1,261 @@
+        )
+
+        return await self.async_step_confirm()
+
+    async def async_step_confirm(
+        self, user_input=None
+    ) -> data_entry_flow.ConfigFlowResult:
+        """Allow the user to confirm adding the device."""
+        if user_input is not None:
+            return self.async_create_entry(
+                title=self.model or DEFAULT_NAME,
+                data={
+                    CONF_HOST: self.host,
+                    CONF_MODEL: self.model,
+                    CONF_SERIAL: self.serial_number,
+                    CONF_NAME: DEFAULT_NAME,
+                },
+                options={
+                    OPTION_INPUT_SOURCES_IGNORE: user_input.get(
+                        OPTION_INPUT_SOURCES_IGNORE
+                    )
+                    or [],
+                    OPTION_INPUT_SOURCES: user_input.get(OPTION_INPUT_SOURCES) or {},
+                },
+            )
+
+        return self.async_show_form(step_id="confirm")
+
+    async def async_step_import(self, import_data: dict) -> ConfigFlowResult:
+        """Import data from configuration.yaml into the config flow."""
+        res = await self.async_step_user(import_data)
+        if res["type"] == FlowResultType.CREATE_ENTRY:
+            _LOGGER.info(
+                "Successfully imported %s from configuration.yaml",
+                import_data.get(CONF_HOST),
+            )
+        elif res["type"] == FlowResultType.FORM:
+            _LOGGER.error(
+                "Could not import %s from configuration.yaml",
+                import_data.get(CONF_HOST),
+            )
+        return res
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(
+        config_entry: ConfigEntry,
+    ) -> OptionsFlow:
+        """Return the options flow."""
+        return YamahaOptionsFlowHandler(config_entry)
+
+
+class YamahaOptionsFlowHandler(OptionsFlow):
+    """Handle an options flow for Yamaha."""
+
+    def __init__(self, config_entry: ConfigEntry) -> None:
+        """Initialize options flow."""
+        self._input_sources_ignore: list[str] = config_entry.options[
+            OPTION_INPUT_SOURCES_IGNORE
+        ]
+        self._input_sources: dict[str, str] = config_entry.options[OPTION_INPUT_SOURCES]
+
+    async def async_step_init(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Manage the options."""
+        yamaha = self.hass.data[DOMAIN][self.config_entry.entry_id]
+        inputs = await self.hass.async_add_executor_job(yamaha.inputs)
+
+        if user_input is not None:
+            sources_store: dict[str, str] = {
+                k: v for k, v in user_input.items() if k in inputs and v != ""
+            }
+
+            return self.async_create_entry(
+                data={
+                    OPTION_INPUT_SOURCES: sources_store,
+                    OPTION_INPUT_SOURCES_IGNORE: user_input.get(
+                        OPTION_INPUT_SOURCES_IGNORE
+                    ),
```

Comment:
For the ignore one, I believe there is a way on the device to limit the sources, can explore that option?

Reply:
Same here not on my device (RX2020) there is no functionality to support it in the rxv lib.

Reply:
If you're referring to the YNCA protocol, it does a better job at returning the renamed inputs from the device, but you can't store or retrieve ignored inputs. (based on the protocol documentation found at https://www.sdu.se/pub/yamaha/yamaha-ynca-receivers-protocol.pdf)

Reply:
Unfortunately I didn't find anything like that on my device menus or in the manual (https://europe.yamaha.com/files/download/other_assets/7/324637/RX-A2010_om_En2.pdf), so I'll resolve this for now.
If it's somewhere well hidden do let me know :-)